commons/src/main/java/fr/free/nrw/commons/category/PrefixUpdater.java

```diff
@@ -0,0 +1,80 @@
+package fr.free.nrw.commons.category;
+
+import android.os.AsyncTask;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+
+import org.mediawiki.api.ApiResult;
+import org.mediawiki.api.MWApi;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import fr.free.nrw.commons.CommonsApplication;
+
+public class PrefixUpdater extends AsyncTask<Void, Void, ArrayList<String>> {
+    //TODO: Return type?
+
+    private String filter;
+    private static final String TAG = PrefixUpdater.class.getName();
+    CategorizationFragment catFragment;
+
+    public PrefixUpdater(CategorizationFragment catFragment) {
+        this.catFragment = catFragment;
+    }
+
+    @Override
+    protected void onPreExecute() {
+        super.onPreExecute();
+        filter = catFragment.categoriesFilter.getText().toString();
+        catFragment.categoriesSearchInProgress.setVisibility(View.VISIBLE);
+        catFragment.categoriesNotFoundView.setVisibility(View.GONE);
+
+        catFragment.categoriesSkip.setVisibility(View.GONE);
+    }
+
+    @Override
+    protected void onPostExecute(ArrayList<String> categories) {
+        super.onPostExecute(categories);
+        catFragment.setCatsAfterAsync(categories, filter);
+        //TODO: Return its own List
```

Comment:
Oof, sorry. I forgot to remove the TODOs, they were outdated. You're right, subclassing is necessary. Not doing that was the reason for how my old branch got so messed up.